pulsar/consumer_partition.go

```diff
@@ -1440,17 +1452,18 @@ func (pc *partitionConsumer) dispatcher() {
 			}
 			nextMessageSize = queueMsg.size()
 
-			if pc.dlq.shouldSendToDlq(&nextMessage) {
-				// pass the message to the DLQ router
-				pc.metrics.DlqCounter.Inc()
-				messageCh = pc.dlq.Chan()
-			} else {
-				// pass the message to application channel
-				messageCh = pc.messageCh
+			if !pc.isSeeking.Load() {
```

Comment:
Why not use channel for communication here? but use memory.

Reply:
The channel is used to add or consume once. There is a loop, so we need to use `atomic.Bool` instead `chan`.